extensions/cornerstone-dicom-sr/src/utils/hydrateStructuredReport.ts

```diff
@@ -191,13 +197,21 @@ export default function hydrateStructuredReport(
         // StudyInstanceUID,
       } = instance;
 
+      const activeViewportId = viewportGridService.getActiveViewportId();
+      const cornerstoneViewport =
+        cornerstoneViewportService.getCornerstoneViewport(activeViewportId);
+      const imageIds = cornerstoneViewport.getImageIds();
+      const viewRef = cornerstoneViewport.getViewReference({
+        sliceIndex: imageIds.indexOf(imageId),
+      });
```

Comment:
This is tricky, there are some weird scenarios that we don't want this here, did you get this logic from somewhere else?

Reply:
no, I just wrote it, so if it's not the best then I'm not sure what alternative we have

Reply:
The problem is that the slice index may not be the image id index - what should be passed in is the image id itself, and let the different scenarios be computed inside.

Reply:
Agree

Reply:
That sounds good to me too, I tried that initially however I think the method doesn’t take imageId unless I did something wrong.

I think I need to update it to work on cs3d side, it doesn’t let you get a viewref from an imageId